### ISSUE TITLE
Comet hangs and stops the training run

### ISSUE THREAD

**Issue by depthwise** [comet-ml/issue-tracking#297]
Comet hung for me several times over the past couple of days. 

Symptom: it issues this request endlessly `I0904 11:26:00.352577 140522752837376 connectionpool.py:393] https://www.comet.ml:443 "POST /clientlib/status-report/update HTTP/1.1" 200 123` and all other activity (as well as training) stops indefinitely.

This is Comet 2.0.11, training is in eager Keras using TF 2.0.0-rc0. I don't know if it's related or not. Others on the team report similar issues both on-prem and on AWS.

**Comment by depthwise** [comment]
I'm currently running the same code without Comet logging to see if it's something in our code, but the few runs I've already done completed without any issues. Is this a know issue? How do I diagnose further?

**Comment by dsblank** [comment]
@depthwise Can you try comet_ml version 2.0.12. We introduced support for TF2rc1 v2 in this version. If that doesn't help, can you send me a Comet debug logging file? See: https://www.comet.ml/docs/python-sdk/advanced/#trouble-shooting

doug@comet.ml

**Comment by depthwise** [comment]
I've just started another run with 2.0.12, and I'll start a few more overnight. I'll let you know how that goes.

**Comment by dsblank** [comment]
Thanks! You can also send the results to support@comet.ml, our new issue tracking email-based system. It provides a better environment for follow-ups and keeping private information protected.

**Comment by dsblank** [comment]
Also, another test you can try: if you swap out Experiment with OfflineExperiment, you might get a zip file that we can further diagnose, for example, if it is attempting to send too much data. One difference is that we will log all scalars and histograms in TF2. For more info on OfflineExperiment, see: https://www.comet.ml/docs/python-sdk/OfflineExperiment/#offlineexperiment__init__ and make sure you have set `offline_directory` there or in the config as COMET_OFFLINE_DIRECTORY: https://www.comet.ml/docs/python-sdk/advanced/#comet-configuration-variables

**Comment by depthwise** [comment]
I don't use any of the automatic logging stuff. I send only a few metrics: learning rate and training loss every 10 batches (at about 4 batches per second, so basically 2 floats every 2.5 seconds), and 4 scalar validation metrics at the end of each epoch. Epochs are about 3 minutes each.

**Comment by dsblank** [comment]
>  I send only a few metrics: learning rate and training loss every 10 batches (the throughput is about 4 batches per second, so basically 2 floats every 2.5 seconds), and 4 scalar validation metrics at the end of each epoch. Epochs are about 3 minutes each.

And that is being logged correctly?

**Comment by depthwise** [comment]
Seems like it is being logged correctly. I'm not running all that many experiments in parallel, usually up to 4.

**Comment by dsblank** [comment]
It sounds like you are at the cutting edge, and all that is designed to work so far is working (except for that hanging issue). If you figure out how to get the graph from your TF2 model or what causes the strange hanging issue, please let us know.

**Comment by dsblank** [comment]
Also, I was reminded in the comet user bug slack channel (https://cometml.slack.com/messages/CA32RGA1L) that if your code has a lot of output (or is in a git repo with lots of unchecked-in changes) that the amount of data sent could be quite large (and perhaps cause such a hang). The OfflineExperiment can help diagnose that. Also two other ideas:

1. Turn off some of the logging (such as stdout): https://www.comet.ml/docs/python-sdk/Experiment/#experiment__init__
2. Explore the OfflineExperiment ZIP file: https://www.comet.ml/docs/command-line/#comet-offline

**Comment by depthwise** [comment]
Well, whatever it was, it seems to be gone now with the newest version of the PIP package.

**Comment by dsblank** [comment]
Good to hear, and thanks for updating the issue. Let us know if you have any further questions or issues.
